pkg/apis/deployment/v1alpha/server_group_spec.go

```diff
@@ -110,8 +114,27 @@ func (s ServerGroupSpec) Validate(group ServerGroup, used bool, mode DeploymentM
 				minCount = 2
 			}
 		}
+		var specMinCount int
```

Comment:
Suggest to add helper functions `GetMinCount() int` & `GetMaxCount() int` that return the effective min/max count values.
If the setting is nil, return return 1/maxInt.

Then in Validate use those functions to avoid nil checks in there.